packages/app/src/app/pages/Sandbox/Editor/Navigation/Navigation.tsx

```diff
@@ -44,34 +85,33 @@ export const Navigation = observer(
     } = useSignals();
     const store = useStore();
 
+    const shownItems = getWorkspaceItems(store);
+    const disabledItems = getDisabledItems(store);
+
     return (
       <Container topOffset={topOffset} bottomOffset={bottomOffset}>
-        {getWorkspaceItems(store)
-          .filter(w => !w.show || w.show(store))
-          .map(item => {
-            const { id, name } = item;
-            const Icon = IDS_TO_ICONS[id];
-            const selected =
-              !store.workspace.workspaceHidden &&
-              id === store.workspace.openedWorkspaceItem;
-            return (
-              <Tooltip key={id} placement="right" content={name}>
-                <IconContainer
-                  selected={selected}
-                  onClick={() => {
-                    if (selected) {
-                      setWorkspaceHidden({ hidden: true });
-                    } else {
-                      setWorkspaceHidden({ hidden: false });
-                      setWorkspaceItem({ item: id });
-                    }
-                  }}
-                >
-                  <Icon />
-                </IconContainer>
-              </Tooltip>
-            );
-          })}
+        {shownItems.map(item => (
+          <IconComponent
+            key={item.id}
+            item={item}
+            store={store}
```

Comment:
`store`, `setWorkspaceHidden` & `setWorkspaceItem` can be deleted here (see above).